airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/writer/BaseS3Writer.java

```diff
@@ -130,6 +135,39 @@ protected void closeWhenFail() throws IOException {
     // Do nothing by default
   }
 
+  protected String determineOutputFilename(final S3DestinationConfig config, final S3Format s3Format, final Timestamp uploadTimestamp)
+      throws IOException {
+    final String outputFilename;
+    if (StringUtils.isNotBlank(config.getFileNamePattern())) {
+      outputFilename = getOutputFilename(S3FilenameTemplateParameterObject
+          .builder()
+          .fileExtension(s3Format.getFileExtension())
+          .fileNamePattern(config.getFileNamePattern())
+          .build());
+    } else {
+      outputFilename = getOutputFilename(uploadTimestamp, s3Format);
+    }
+    return outputFilename;
+  }
+
+  protected String determineOutputFilename(final S3DestinationConfig config,
```

Comment:
this looks like too much code duplication to me. As far as I can see, this is the only difference between these two methods:
`outputFilename = getOutputFilename(uploadTimestamp, s3Format);`
vs
`outputFilename = getOutputFilename(uploadTimestamp, suffix, s3Format);`

could you please consolidate these two methods into one. You could do something like:
`outputFilename = suffix == null ? getOutputFilename(uploadTimestamp, s3Format) :  getOutputFilename(uploadTimestamp, suffix, s3Format);`